internal/dag/builder.go

```diff
@@ -38,18 +37,15 @@ type Builder struct {
 	// from which to build a DAG.
 	Source KubernetesCache
 
+	// data holds the dag information after a builder
+	// processes through the Kubernetes cache which is
+	// used to build a new DAG.
+	data BuilderData
```

Comment:
I'm not in love with this name, but `cache` seemed incorrect as well. I'm looking to use this for the builder refactor so each implementation can be in charge of populating this struct.

Reply:
Maybe `BuildContext`, or `BuilderState`?